src/CMakeLists.txt

```diff
@@ -17,12 +17,20 @@ if (GUI)
     if (WIN32)
         list (APPEND QBT_QT_COMPONENTS WinExtras)
     endif(WIN32)
+    if (APPLE)
+        list (APPEND QBT_GUI_OPTIONAL_LINK_LIBRARIES objc)
+        list (APPEND QBT_QT_COMPONENTS MacExtras)
+    endif (APPLE)
 endif (GUI)
 if (DBUS)
     list (APPEND QBT_QT_COMPONENTS DBus)
 endif (DBUS)
 find_package(Qt5 5.5.1 COMPONENTS ${QBT_QT_COMPONENTS} REQUIRED)
 
+if (GUI AND APPLE)
+    # Fix MOC inability to detect macOS
+    add_definitions(-DQ_OS_MAC)
```

Comment:
I can not find the same define for qmake build. How does it build then? Also, if this a fix for a specific file and bug, would it be possible to pass the flag for that file target only supplying a comment why the flag is needed?